### PR TITLE
Playing with dominances

### PR DESCRIPTION
**Pull request by jonsterling**
I'm just opening this draft PR to track my work on dominances, lifting, etc. I'll let you know when it is ready for review.

### CODE REVIEW

**Comment by martinescardo** [comment]
Great. 

Note: in `TypeTopology` we use e.g. `is-defined` rather than the camel case `isDefined` etc.

Second note: there is a lot of stuff about lifting in the folder `Lifting`.

**Comment by martinescardo** [comment]
Any progress with this?

File: source/Dominance/Lifting.lagda [hunk 1]
```diff
@@ -6,74 +6,183 @@ Martin Escardo, January 2018, May 2020
...
+
+  _≅_ : L X → L X → 𝓣 ⊔ 𝓥 ̇
+  (P , u , dP) ≅ (Q , v , dQ) =
+    Σ f ꞉ P ⇔ Q , u ∼ v ∘ pr₁ f
+
+  ≅-refl : (u : L X) → u ≅ u
+  ≅-refl u = (id , id) , λ _ → refl
+
+  -- TODO: move or find in library
+  Σ-assoc-equiv
```
Comment (jonsterling) [inline source/Dominance/Lifting.lagda:82]: @martinescardo Is there somewhere in the library that already has this helper function, or to where you would like me to move it?

File: source/Dominance/Lifting.lagda [hunk 2]
```diff
@@ -6,74 +6,183 @@ Martin Escardo, January 2018, May 2020
...
+   : {𝓥 𝓦 𝓧 : _} {A : 𝓥 ̇} {B : A → 𝓦 ̇} {C : (x : A) → B x → 𝓧 ̇}
+   → (Σ xy ꞉ Σ B , C (pr₁ xy) (pr₂ xy)) ≃ (Σ x ꞉ A , Σ y ꞉ B x , C x y)
```
Comment (martinescardo) [inline source/Dominance/Lifting.lagda:84]: You can write this line as

`→ (Σ (x , y) ꞉ Σ B , C x y) ≃ (Σ x ꞉ A , Σ y ꞉ B x , C x y)`

or simply `→ Σ C ≃ (Σ x ꞉ A , Σ y ꞉ B x , C x y)`.

It is in `UF.EquivalenceExamples`, called `Σ-assoc`.

Comment (jonsterling) [reply source/Dominance/Lifting.lagda]: Thanks!!

**Pull request merged**